packages/cozy-harvest-lib/src/components/hoc/withDomain.jsx

```diff
@@ -0,0 +1,30 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+
+const getGlobalDomain = () => {
+  try {
+    return document.querySelector('[role=application]').dataset.cozyDomain
+  } catch (error) {
+    throw new Error(
+      'Cannot retrieve data-cozy-domain attribute on [role=application] element'
+    )
+  }
+}
```

Comment:
This should be memoized. Here we are querying the DOM at every render. This is very bad performance wise.